### PR TITLE
[MOS-23] Launch a npm package

### PR DESCRIPTION
**Pull request by Alderr**
## Purpose/Description:
Release an npm package to use for our needs in our dashboard. 

- Added aliases or easier importing
- Adjusted package.json for package release
- Added rollup for building our package for releasing 
- Fixing random assortment of TS errors inside Button, theme, and Palette
- Added ts check command
- Prettier fix on almost all files

## Jira Link:
[MOS-23](https://collagegroup.atlassian.net/browse/MOS-23)

[MOS-23]: https://collagegroup.atlassian.net/browse/MOS-23?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -3,4 +3,4 @@
     "tabWidth": 4,
```
Comment (diego20) [inline .prettierrc]: Sure about the tabWidth: 4? It's 2 in Dashboard.

File: src/components/atoms/Button/Button.stories.tsx [hunk 2]
```diff
@@ -1,10 +1,13 @@
-import { ComponentStory, StoryObj, Meta } from '@storybook/react';
```
Comment (diego20) [inline src/components/atoms/Button/Button.stories.tsx:1]: ComponentStory is marked as deprecated. FYI. I leave it to you if you consider it's ok this way or it's worth updating to 'StoryFn'

File: .eslintrc [hunk 3]
```diff
@@ -1,9 +1,15 @@
 {
-  "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended", "prettier"],
-  "parser": "@typescript-eslint/parser",
-  "plugins": ["@typescript-eslint", "prettier"],
-  "root": true,
-  "rules": {
-    "prettier/prettier": "error"
-  }
+    "extends": [
+        "eslint:recommended",
```
Comment (Alderr) [inline .eslintrc]: rules to enable ts + lint checks in the IDE and `npx tsc`

File: .prettierrc (hunk 1 above)
Comment (diego20) [reply .prettierrc]: Can you see this comment? @Alderr

File: .storybook/main.ts [hunk 4]
```diff
@@ -1,32 +1,33 @@
...
-      propFilter: (prop) => (prop.parent ? !/node_modules/.test(prop.parent.fileName) : true),
+    stories: ['../src/**/*.mdx', '../src/**/*.stories.@(js|jsx|ts|tsx)'],
+    addons: [
+        '@storybook/addon-links',
+        '@storybook/addon-essentials',
+        '@storybook/addon-interactions',
+    ],
+    typescript: {
+        check: false,
+        checkOptions: {},
```
Comment (diego20) [inline .storybook/main.ts]: I'm getting an error that this property does not exist in "typescript" object. Remove

File: .storybook/manager.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
-import { addons } from "@storybook/addons";
+import { addons } from '@storybook/addons';
```
Comment (diego20) [inline .storybook/manager.ts]: I'm getting an error "Cannot find module '@storybook/addons' or its corresponding type declarations".
This package does not exist in package.json file.
Looking at documentation the package contents have been moved. https://www.npmjs.com/package/@storybook/addons

File: src/components/atoms/Button/Button.stories.tsx [hunk 6]
```diff
@@ -1,138 +1,140 @@
...
-            </Button>
-            <Button variant="secondary" disabled>
-                Button
-            </Button>
-            <Button variant="tertiary" disabled>
-                Button
-            </Button>
-        </Box>
-    );
+export const Disabled: Story = () => {
```
Comment (diego20) [inline src/components/atoms/Button/Button.stories.tsx]: Story is marked as deprecated. Message suggests using StoryFn or StoryObj

File: src/components/atoms/Button/Button.stories.tsx [hunk 7]
```diff
@@ -1,138 +1,140 @@
...
-    return (
-        <Box display="flex" justifyContent="space-around">
-            <Button color="uiGray">uiGray</Button>
-            <Button color="collageViolet">collageViolet</Button>
-            <Button color="collageOrange">collageOrange</Button>
-            <Button color="collageTeal">collageTeal</Button>
-            <Button color="collageYellow">collageYellow</Button>
-        </Box>
-    );
+export const Colors: Story = () => {
```
Comment (diego20) [inline src/components/atoms/Button/Button.stories.tsx]: Story is marked as deprecated. Message suggests using StoryFn or StoryObj

File: src/components/atoms/Button/Button.tsx [hunk 8]
```diff
@@ -1,172 +1,206 @@
...
-            {children}
-        </ButtonRoot>
-    );
-};
+function Button({
+  children,
+  variant = 'primary',
+  size = 'medium',
+  // color is a custom color prop we use but MUI does not so I destruct-ed it out.
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (diego20) [inline src/components/atoms/Button/Button.tsx]: I think this linter comment can be removed. The color prop is being used

File: vite.config.ts [hunk 9]
```diff
@@ -1,7 +1,20 @@
 import { defineConfig } from 'vite';
 import react from '@vitejs/plugin-react';
+import resolve from '@rollup/plugin-node-resolve';
```
Comment (diego20) [inline vite.config.ts]: None of these 4 imports nor the required package.json are being used. Remove

**Review by diego20 — COMMENTED** [review]
In general, as far as I'm aware the React import is no longer required in React 17+. Not demanding to remove them all but maybe we should agree on leaving it or not for consistency?
Also, tried running the lint command and it failed, maybe also worth discussing if / when it should be fixed? In the same train of thought also adding linter check to GH actions?

**Comment by Alderr** [comment]
@diego20 Added GitHub Actions to run checks for future PRs. I've addressed the comments as well.

**Review by diego20 — APPROVED** [review]
LGTM

**Pull request merged**